src/externalized/mercs/MERCListingModel.cc

```diff
@@ -36,24 +36,51 @@ static SpeckQuoteType SpeckQuoteTypefromString(std::string s)
 	throw std::runtime_error("unsupported quote type: " + s);
 }
 
-MERCListingModel* MERCListingModel::deserialize(uint8_t index, const rapidjson::Value& json)
+MERCListingModel* MERCListingModel::deserialize(uint8_t index, const rapidjson::Value& json, const MercSystem* mercSystem)
 {
+	ST::string profileName = json["profile"].GetString();
+	auto profile = mercSystem->getMercProfileInfoByName(profileName);
+	if (profile == NULL) {
+		throw std::runtime_error(ST::format("'{}' does not refer to a valid profile", profile).c_str());
```

Comment:
Should we use `DataError` here and below?

Reply:
Fixed.